nemo/collections/asr/models/classification_models.py

```diff
@@ -468,51 +654,149 @@ def change_labels(self, new_labels: List[str]):
 
         Args:
+            labels: The current labels of the model
             cfg: The config of the decoder which will be updated.
         """
         OmegaConf.set_struct(cfg, False)
 
-        labels = self.cfg.labels
-
         if 'params' in cfg:
             cfg.params.num_classes = len(labels)
         else:
             cfg.num_classes = len(labels)
 
         OmegaConf.set_struct(cfg, True)
 
-    def _setup_transcribe_dataloader(self, config: Dict) -> 'torch.utils.data.DataLoader':
+
+class EncDecRegressionModel(_EncDecBaseModel):
+    """Encoder decoder class for speech regression models.
+    Model class creates training, validation methods for setting up data
+    performing model forward pass.
+    """
+
+    @classmethod
+    def list_available_models(cls) -> List[PretrainedModelInfo]:
         """
-        Setup function for a temporary data loader which wraps the provided audio file.
+        This method returns a list of pre-trained model which can be instantiated directly from NVIDIA's NGC cloud.
+        Returns:
+            List of available pre-trained models.
+        """
+        result = []
+
+        return result
+
+    def __init__(self, cfg: DictConfig, trainer: Trainer = None):
+        if not cfg.get('is_regression_task', False):
+            raise ValueError(f"EndDecRegressionModel requires the flag is_regression_task to be set as true")
+        super().__init__(cfg=cfg, trainer=trainer)
+
+    def _setup_preprocessor(self):
+        return EncDecRegressionModel.from_config_dict(self._cfg.preprocessor)
+
+    def _setup_encoder(self):
+        return EncDecRegressionModel.from_config_dict(self._cfg.encoder)
+
+    def _setup_decoder(self):
+        return EncDecRegressionModel.from_config_dict(self._cfg.decoder)
+
+    def _setup_loss(self):
+        return MSELoss()
+
+    def _setup_metrics(self):
+        self._mse = MeanSquaredError()
+        self._mae = MeanAbsoluteError()
+
+    @property
+    def output_types(self) -> Optional[Dict[str, NeuralType]]:
+        return {"preds": NeuralType(tuple('B'), RegressionValuesType())}
+
+    @typecheck()
+    def forward(self, input_signal, input_signal_length):
+        logits = super().forward(input_signal=input_signal, input_signal_length=input_signal_length)
+        return logits.view(-1)
+
+    # PTL-specific methods
+    def training_step(self, batch, batch_idx):
+        self.training_step_end()
+        audio_signal, audio_signal_len, targets, targets_len = batch
+        logits = self.forward(input_signal=audio_signal, input_signal_length=audio_signal_len)
+        loss = self.loss(preds=logits, labels=targets)
+        train_mse = self._mse(preds=logits, target=targets)
+        train_mae = self._mae(preds=logits, target=targets)
+
+        tensorboard_logs = {
+            'train_loss': loss,
+            'train_mse': train_mse,
+            'train_mae': train_mae,
+            'learning_rate': self._optimizer.param_groups[0]['lr'],
+        }
+
```

Comment:
@fayejf Could you send a PR to change use self.log(tensorboard_logs) and remove the 'log' part in the return below, for the train step? Thanks.

Reply:
Should I use self.log(tensorboard_logs)?, it's asking for a key value. Maybe we should do:

`  for key in tensorboard_logs:
            self.log(f'{key}', tensorboard_logs[key], prog_bar=True)`

Reply:
There's self.log_dict(). It's fine, we already merged it and it shoupd be fine for now

Reply:
@titu1994  Sure!